File: backend/jnstore-sboot-auth/src/main/java/br/com/jnstore/sboot/auth/service/PerfilService.java
package br.com.jnstore.sboot.auth.service;

import br.com.jnstore.sboot.auth.domain.TbPerfil;
import org.springframework.security.core.GrantedAuthority;

import java.util.List;

public interface PerfilService {
    TbPerfil criarPerfil(TbPerfil tbPerfil);

    boolean podeEditarProduto(List<GrantedAuthority> grants);
    boolean podeVisualizarProduto(List<GrantedAuthority> grants);
    boolean podeVisualizarValorCompra(List<GrantedAuthority> grants);
    boolean podeVisualizarCategoria(List<GrantedAuthority> grants);
    boolean podeEditarCategoria(List<GrantedAuthority> grants);
    boolean podeAbrirCaixa(List<GrantedAuthority> grants);
    boolean podeVisualizarCaixa(List<GrantedAuthority> grants);
    boolean podeFecharCaixa(List<GrantedAuthority> grants);
    boolean podeRetirarCaixa(List<GrantedAuthority> grants);
    boolean podeEditarVenda(List<GrantedAuthority> grants);
    boolean podeVisualizarVenda(List<GrantedAuthority> grants);
    boolean podeEditarValorVenda(List<GrantedAuthority> grants);
}

File: frontend/jnstore-angular/src/app/feature/services/perfil.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from '../../../environments/environment';
import { Observable } from 'rxjs';
import { PerfilRepresentation } from '../models';

@Injectable({ providedIn: 'root' })
export class PerfilService {
  private baseUrl = environment.apiUrl || 'http://localhost:8080';

  constructor(private http: HttpClient) {}

  criar(perfil: PerfilRepresentation): Observable<PerfilRepresentation> {
      return this.http.post<PerfilRepresentation>(`${this.baseUrl}/api/profiles`, perfil);
  }

  podeEditarProduto(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeEditarProduto`);
  }

  podeVisualizarProduto(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeVisualizarProduto`);
  }

  podeVisualizarValorCompra(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeVisualizarValorCompra`);
  }

  podeVisualizarCategoria(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeVisualizarCategoria`);
  }

  podeEditarCategoria(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeEditarCategoria`);
  }

  podeAbrirCaixa(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeAbrirCaixa`);
  }

  podeVisualizarCaixa(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeVisualizarCaixa`);
  }

  podeFecharCaixa(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeFecharCaixa`);
  }

  podeRetirarCaixa(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeRetirarCaixa`);
  }

  podeEditarVenda(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeEditarVenda`);
  }

  podeVisualizarVenda(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeVisualizarVenda`);
  }

  podeEditarValorVenda(): Observable<boolean> {
      return this.http.get<boolean>(`${this.baseUrl}/api/profiles/podeEditarValorVenda`);
  }
}
